src/prted/pmix/pmix_server_register_fns.c

```diff
@@ -308,6 +308,7 @@ int prrte_pmix_server_register_nspace(prrte_job_t *jdata)
     /* register any psets for this job */
     for (i=0; i < (int)jdata->num_apps; i++) {
         app = (prrte_app_context_t*)prrte_pointer_array_get_item(jdata->apps, pptr->app_idx);
```

Comment:
Wait - this doesn't make sense. pptr->app_idx is fixed - this should be looking at the ith location. The loop should technically go until jdata->apps->size, though I admit that the pointer array _should_ be left-packed.

Reply:
For this, I made a change that all procs are now constructed with app->idx as a UINT32_MAX instead of 0.
It seems that the prted (or I am not sure?) can be inserted into the node->proc structure, which necessitated this change.

Reply:
Not sure I understand your answer - what does the default app idx have to do with it? You are trying to loop over all apps in this particular job, and the index passed to "get_item" isn't doing that.

Reply:
I reverted it back - checking jobid + app idx works.